CONTRIBUTING.md

```diff
@@ -31,3 +31,32 @@ In general, here's a suggested workflow:
 6. Actively participate in your pull request.
    If the maintainers ask for a revision to your proposed changes, make it promptly to help get your PR merged as soon as possible.
    Inactive PR's or Issues may be marked as Deferred or closed if no response is given.
+
+## Branching/Versioning Scheme
+
+***Note: This guide is intended mainly for core contributors.***
+
+Dot uses [SemVer v2](http://semver.org/spec/v2.0.0.html) as a model to standardize our versioning system.
+A "breaking change" in our context is something that might cause a user to have to change their personal config; any change to the "API" (which is the set of functions exposed to the user) is considered as such.
+
+In general, our branching workflow consists of two persistent branches:
+
+- `master`, which contains all of the latest changes and should be in a stable state.
+  This is used as the base for all of the `release-` branches (see below.)
+- `stable`, which is a tracking branch that gets releases merged into it.
+  This is the branch to be used by users who only want stable features to be added to their configuration.
+
+In addition, we use the following scheme for additional branches:
+
+- `release-vX.Y` and `release-vX.Y.Z` branches are branches used to control versioning.
```

Comment:
Bumping a version is something that should probably not really be done by people submitting changes.  That's project-level work that should really just be done by the maintainers when they know what they want to merge.  We'd delete the branch, but people would be able to create the branch again if we needed to add something to a previous version in a patch release.